contrib/node/src/python/pants/contrib/node/subsystems/mixins.py

```diff
@@ -0,0 +1,78 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.contrib.node.subsystems.command import command_gen
+
+
+PACKAGE_MANAGER_NPM = 'npm'
+PACKAGE_MANAGER_YARNPKG = 'yarnpkg'
+
+
+class PackageManagerMixin(object):
+  """Defines node package manager functionalities."""
+
+  def _get_installation_args(self, install_optional, production_only, force):
+    """Returns command line args for installing package.
+
+    :param install_optional: True to request install optional dependencies.
+    :param production_only: True to only install production dependencies, i.e.
+      ignore devDependencies.
+    :param force: True to force re-download dependencies.
+    :rtype: list of strings
+    """
+    raise NotImplementedError
+
+  def _get_run_script_args(self):
+    """Returns command line args to run a package.json script.
+
+    :rtype: list of strings
+    """
+    raise NotImplementedError
+
+  def install_packages(
```

Comment:
Can we add --frozen-lockfile here as well? This will not generate a lock-file and fail if an update is detected.